File: src/main/java/com/rcoverzola/ejers/HowOldAreYou.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.rcoverzola.ejers;

import java.util.Scanner;

/**
 *
 * @author Roger Coverzola Bogado
 */
public class HowOldAreYou {
    private int OLD_AGE = 18;
    private int ZERO = 0;
    private int MAX_AGE = 150;
    private String[] MESSAGES = {"Cuantos años tienes?: ","Mayor","Menor", "Ingresar solo el numeros que correspondan a tu edad","No es un valor valido!!.","Error: MENSAJES no puede estar vacío", "Adios!!"};
    private Scanner scanner;
    private boolean EXIT = true;
      
    public HowOldAreYou() {
       this.scanner = new Scanner(System.in);
    }
    
    public int validateAge(int age) {
        if (age<=this.ZERO || age>= this.MAX_AGE) {
            System.out.println(this.MESSAGES[4]);
            this.EXIT = false;
            this.scanner.close();
             System.out.println(this.MESSAGES[6]);
        }
            return age;
        
    }
   
    
    private void isOld(int age){
        if(this.OLD_AGE <= age){
            System.out.println("Eres "+this.MESSAGES[1]+" de edad!");
        }else{
            System.out.println("Eres "+this.MESSAGES[2]+" de edad!");
        }
    }
    
    public void ask(){
        
        while (this.EXIT) {
        System.out.println(this.MESSAGES[3]);
        System.out.println(this.MESSAGES[0]);
        
        Integer age = validateAge(this.scanner.nextInt());
        
        if (this.EXIT){
            isOld(age);
        }
        }
        
    }
}

File: src/HowOldAreYou.ts
export class HowOldAreYou {
    private static readonly OLD_AGE: number = 18;
    private static readonly ZERO: number = 0;
    private static readonly MAX_AGE: number = 150;
    private static readonly MESSAGES: string[] = [
        "Cuantos años tienes?: ",
        "Mayor",
        "Menor", 
        "Ingresar solo el numeros que correspondan a tu edad",
        "No es un valor valido!!.",
        "Error: MENSAJES no puede estar vacío", 
        "Adios!!"
    ];
    private scanner: any;
    private EXIT: boolean = true;
      
    constructor() {
        this.scanner = require('readline').createInterface({
            input: process.stdin,
            output: process.stdout
        });
    }
    
    public validateAge(age: number): number {
        if (age <= HowOldAreYou.ZERO || age >= HowOldAreYou.MAX_AGE) {
            console.log(HowOldAreYou.MESSAGES[4]);
            this.EXIT = false;
            this.scanner.close();
            console.log(HowOldAreYou.MESSAGES[6]);
        }
        return age;
    }
   
    private isOld(age: number): void {
        if (HowOldAreYou.OLD_AGE <= age) {
            console.log("Eres " + HowOldAreYou.MESSAGES[1] + " de edad!");
        } else {
            console.log("Eres " + HowOldAreYou.MESSAGES[2] + " de edad!");
        }
    }
    
    public ask(): void {
        const askQuestion = () => {
            if (!this.EXIT) return;
            
            console.log(HowOldAreYou.MESSAGES[3]);
            console.log(HowOldAreYou.MESSAGES[0]);
            
            this.scanner.question('', (input: string) => {
                const age: number = this.validateAge(parseInt(input));
                
                if (this.EXIT) {
                    this.isOld(age);
                }
                
                if (this.EXIT) {
                    askQuestion();
                }
            });
        };
        
        askQuestion();
    }
}
